crates/gpui/src/platform/linux/wayland/client.rs

```diff
@@ -1524,6 +1526,8 @@ impl Dispatch<wl_pointer::WlPointer, ()> for WaylandClientStatePtr {
                         let click_elapsed = state.click.last_click.elapsed();
 
                         if click_elapsed < DOUBLE_CLICK_INTERVAL
+                            && state.click.last_mouse_button.is_some()
+                            && state.click.last_mouse_button.unwrap() == button
```

Comment:
I think this is better, that way we can avoid an unwrap.
```suggestion
                        && state.click.last_mouse_button.is_some_and(|prev_button| prev_button == button)
```

Reply:
Thanks for the suggestion. I updated the branch accordingly.